modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -182,6 +186,25 @@ object CurrencySnapshotAcceptanceManager {
           .getOrElse(acc)
       }
 
+    private def validateFeeTxs(
+      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
+    ): F[Unit] =
+      OptionT
+        .fromOption[F](maybeTxs)
+        .flatMap { txs =>
+          OptionT
+            .fromOption(NonEmptyList.fromList(txs.toList))
+            .semiflatMap(feeTransactionValidator.validateMultiple)
+            .flatMapF {
+              case Validated.Valid(_) => ().pure[F].map(_.some)
+              case Validated.Invalid(errors) =>
+                new Exception(s"FeeTransaction validation failed: ${errors}")
+                  .raiseError[F, Option[Unit]]
+            }
+        }
+        .value
+        .void
```

Comment:
do we have such a construction already in the codebase? as I look on it, it does not make me comfortable 🤔

Reply:
Updated